srcCppCommon/ParameterManager.cxx

```diff
@@ -338,6 +338,38 @@ void ParameterManager::initialize()
             | Middleware::RTIDDSMICRO);
     create("threadPriorities", threadPriorities);
 
+  #ifdef RTI_FLATDATA_AVAILABLE
```

Comment:
I don't see a problem allowing the creation of these parameters out of `RTI_FLATDATA_AVAILABLE`

Reply:
It would not make sense to allow the user to use the parameter `-flatdata` if their DDS version does not provide FlatData support.

Reply:
Agree. :+1: